### ISSUE TITLE
babel-core 5.7 causes false "infix operator" errors

### ISSUE THREAD

**Issue by gpleiss** [babel/babel-eslint#149]
When I updated `node_modules` on my project (which bumped `babel-core` from 5.6.15 to 5.7.2, I started getting "Infix operators must be spaced" linting errors on this chunk of code:

```js
    var title = this.props.title ? (
      <div className="panel-header">
        <h5 className="panel-title-alt">{this.props.title}</h5>
      </div>
    ) : null;
```

**Comment by sebmck** [comment]
@hzoo This might be an issue with tokens in the new parser, let me know if you investigate and find anything, I'll try to when I get the time.

**Comment by gpleiss** [comment]
Sounds good, will do!

And my mistake - I originally said `babel-loader` was causing the error. It's actually `babel-core`.

**Comment by hzoo** [comment]
Will check it out when I'm home! Maybe sbould run all tests with new version to see if any are failing

**Comment by hzoo** [comment]
@sebmck Ok reduced it to `true ? (true) : false;` which errors but `true ? true : false;` doesn't error.

Looks like I'm getting 2 colons (`:`) from the parser when there's the additional parens

```
[ { type: 'Boolean',
    value: 'true',
    start: 0,
    end: 4,
    loc: { start: [Object], end: [Object] },
    range: [ 0, 4 ] },
  { type: 'Punctuator',
    value: '?',
    start: 5,
    end: 6,
    loc: { start: [Object], end: [Object] },
    range: [ 5, 6 ] },
  { type: 'Punctuator',
    value: '(',
    start: 7,
    end: 8,
    loc: { start: [Object], end: [Object] },
    range: [ 7, 8 ] },
  { type: 'Boolean',
    value: 'true',
    start: 8,
    end: 12,
    loc: { start: [Object], end: [Object] },
    range: [ 8, 12 ] },
  { type: 'Punctuator',
    value: ')',
    start: 12,
    end: 13,
    loc: { start: [Object], end: [Object] },
    range: [ 12, 13 ] },
  { type: 'Punctuator', //---------------------- here
    value: ':',
    start: 14,
    end: 15,
    loc: { start: [Object], end: [Object] },
    range: [ 14, 15 ] },
  { type: 'Punctuator', //---------------------- here
    value: ':',
    start: 14,
    end: 15,
    loc: { start: [Object], end: [Object] },
    range: [ 14, 15 ] },
  { type: 'Boolean',
    value: 'false',
    start: 16,
    end: 21,
    loc: { start: [Object], end: [Object] },
    range: [ 16, 21 ] },
  { type: 'Punctuator',
    value: ';',
    start: 21,
    end: 22,
    loc: { start: [Object], end: [Object] },
    range: [ 21, 22 ] } ]
```

**Comment by sebmck** [comment]
Ah, this is due to changes made for babel/babel#1991 then. Thanks for looking into it! I'll try and get a patch out ASAP.

**Comment by AlexKVal** [comment]
The same issue https://github.com/eslint/eslint/issues/3016
I hope it'll add more info :cherries:

**Comment by sebmck** [comment]
@AlexKVal I know the cause. I'm rewriting large parts of the parser that should fix this.

**Comment by hzoo** [comment]
Ok I don't see the issue in babel master so should be fixed in babel `5.8.0`

**Comment by sebmck** [comment]
This should be fixed in >5.8.0.
